### PR TITLE
Dispatch API results as they come in instead of waiting for the slowest one (TEACH-705)

### PR DESCRIPTION
**Pull request by etaderhold**
The dashboard homepage calls `asyncLoadSectionData`, which makes three API calls to retrieve information about sections, valid course offerings, and allowed participant types. This function currently waits for all three API calls to complete before dispatching any of the results to our redux state. This causes the spinner blocking the sections table to run for an unnecessarily long time when the sections API call finishes faster than the others (which is often the case).

In this PR I move the `dispatch` events outside of the `Promise.all.then` callback so that they each run when their respective API call finishes, instead of waiting for the slowest one.

I then switch the spinner condition in the `TeacherSections` component to wait for the `sectionsAreLoaded` boolean to be populated, rather than the `asyncLoadComplete` boolean that is set once all three API calls complete.

The result of this change is that the teacher should see their sections table appear sooner much of the time.

## Links
[JIRA](https://codedotorg.atlassian.net/browse/TEACH-705)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/teacherDashboard/teacherSectionsRedux.js [hunk 1]
```diff
@@ -323,7 +323,7 @@ export const beginCreatingSection = (
   unitId,
   participantType
 ) => ({
-  type: CREATE_SECION_BEGIN,
+  type: CREATE_SECTION_BEGIN,
```
Comment (lfryemason) [inline apps/src/templates/teacherDashboard/teacherSectionsRedux.js:326]: Nice!

**Review by lfryemason — APPROVED** [review]
This looks great. Much cleaner and hopefully faster

**Comment by dju90** [comment]
Do we have a rough ballpark of how fast the section table now loads compared to before?

**Comment by etaderhold** [comment]
> Do we have a rough ballpark of how fast the section table now loads compared to before?

Depends how many sections you have! The more sections you have, the longer it takes to load the info about your sections. For most people this API call is faster than the API call to get the names of all available courses (which is needed to put the course names in the table). However since we aren't waiting for this course names call anymore before rendering the rest of the info in the table, the speedup will be the difference in response times between these two API calls. Could be a couple of seconds for someone with a small number of sections, or no speedup at all for someone with dozens of sections.

**Pull request merged**